core/internal/api/credentials.go

```diff
@@ -15,38 +22,48 @@ type CredentialProvider interface {
 	Apply(req *http.Request) error
 }
 
+// NewCredentialProvider creates a new credential provider based on the SDK
+// settings. Settings for JWT authentication are prioritized above API key
+// authentication.
 func NewCredentialProvider(
 	settings *settings.Settings,
+	logger *slog.Logger,
```

Comment:
we setup the logger to write to a log file, i think if you just use a general logger here it would be an issue no?

Reply:
What's the correct logger to be using?

Reply:
the original that is being passed i think, but you can test it... try adding some log message (info) and see where it is being redirected, if this logger is correct it should be in the runs folder under logs/debug-interal.log

Reply:
double checked - that's where it goes :)